ec/src/group/wnaf.rs

```diff
@@ -0,0 +1,39 @@
+extern crate alloc;
+use crate::Group;
+use alloc::vec::Vec;
+
+pub fn wnaf_mul<G: Group>(table: &[G], scalar_wnaf: &[i64]) -> G {
+    let mut result = G::zero();
+
+    let mut found_one = false;
+
+    for n in scalar_wnaf.iter().rev() {
+        if found_one {
+            result = result.double();
+        }
+
+        if *n != 0 {
+            found_one = true;
+
+            if *n > 0 {
+                result += &table[(n / 2) as usize];
+            } else {
+                result -= &table[((-n) / 2) as usize];
+            }
+        }
+    }
+
+    result
+}
+
+pub fn wnaf_table<G: Group>(table: &mut Vec<G>, mut base: G, window: usize) {
```

Comment:
Should we change this method to just output the table, and not take in a table as input. I'm not seeing any benefit from having it be a mutative argument?

Reply:
Makes sense, modified :+1: